### PR TITLE
reverse the dependency between helix-tui and helix-view

### PR DESCRIPTION
**Pull request by Kethku**
Does step one of https://github.com/helix-editor/helix/issues/39#issuecomment-866446894

"helix-view needs to be refactored to have it's own generic key/style/color structs. Fix previous broken refactor key into helix-view #345 is a start but it depends on crossterm. Using generic structs would let us drop the crossterm dependency from helix-view."

I did this by lifting CursorKind, Margin, Rect, Color, Modifier, and Style into helix-view. I then reversed the dependency making helix-tui depend on helix-view rather than the other way around. I also made helix-view's dependency on crossterm optional in order to allow impl From<<crossterm::style::Color>> on the generic Color type.

Your comment suggested that the lifted types should be "generic". I think the types in the tui crate were pretty generic, but if you have ideas for how these should be modified to make them more "generic", I'd be happy to do it. I just didn't know what you were thinking.

Let me know if this is what you had in mind, it didn't take too long to do, and I'm looking to do these types of refactorings as a way to get to know the codebase better, so I'm not at all attached to this particular PR. Just trying to pitch in.

### CODE REVIEW

**Comment by archseer** [comment]
Note: this probably conflicts #345. Could you maybe copy over the changes from there so we subsume that PR?

I think this is a pretty good idea. It initially wasn't possible because we depended on `tui-rs` externally, but now that it's forked as `helix-tui` it's doable. I have a crossterm->termwiz port in progress that will remove even more code from `helix-tui` so we'll likely be able to drop the crate completely: https://github.com/helix-editor/helix/tree/termwiz

**Comment by Kethku** [comment]
Yes I will copy the changes from #345 over to this PR sometime tomorrow as its late today.

**Comment by archseer** [comment]
No rush :)

**Comment by Kethku** [comment]
Rebased on the merged keybinding PR. Also fixed a bunch of tests. This is ready for review

**Comment by Kethku** [comment]
I had to pull the KeyCode and KeyModifiers types out of crossterm to be compatible with the previous PR. I did the same thing I did with the tui types and added from and into implementations to go back and forth between the originals.

File: helix-view/Cargo.toml [hunk 1]
```diff
@@ -9,20 +9,18 @@ categories = ["editor"]
 repository = "https://github.com/helix-editor/helix"
 homepage = "https://helix-editor.com"
 
-# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
-
 [features]
-term = ["tui", "crossterm"]
-default = ["term"]
+default = []
+term = ["crossterm"]
```
Comment (archseer) [inline helix-view/Cargo.toml:14]: 👍🏻

**Review by archseer — APPROVED** [review]
This is great, thanks!

**Pull request merged**